pkg/discovery/processor/cluster_processor.go

```diff
@@ -74,26 +86,25 @@ func (processor *ClusterProcessor) connectToNodes() error {
 	for _, node := range nodeList {
 		nodeCpuFrequency, err := checkNode(node, processor.nodeScrapper)
 		if err != nil {
-			connectionErrors = append(connectionErrors, fmt.Sprintf("%s:%s", node.Name, err))
+			connectionErrors = append(connectionErrors, fmt.Sprintf("%s [err:%s]", node.Name, err))
+			unreachable = append(unreachable, node.Name)
 		} else {
 			connectionMsgs = append(connectionMsgs,
 				fmt.Sprintf("%s [cpu:%v MHz]", node.Name, nodeCpuFrequency))
 		}
 	}
 
-	if len(connectionMsgs) > 0 {
-		glog.V(2).Infof("Successfully connected to nodes : %s\n", strings.Join(connectionMsgs, ", "))
-	}
-
-	errStr := strings.Join(connectionErrors, ", ")
-	if len(connectionErrors) > 0 {
+	// All nodes are unreachable
+	if len(unreachable) == len(nodeList) {
+		errStr := strings.Join(connectionErrors, ", ")
 		glog.Errorf("Errors connecting to nodes : %s\n", errStr)
+		validationResult.IsValidated = false
+		return validationResult, fmt.Errorf(errStr)
 	}
```

Comment:
should also log the _connectionErrors_ in cases of `len(unreachable) != len(nodeList)`